pkg/operator/ceph/object/zone/controller.go

```diff
@@ -233,10 +237,19 @@ func (r *ReconcileObjectZone) createCephZone(zone *cephv1.CephObjectZone, realmN
 		return reconcile.Result{}, errors.Wrap(err, "failed to parse `radosgw-admin zonegroup get` output")
 	}
 
-	// create zone
+	// create/update zone
 	_, err = object.RunAdminCommandNoMultisite(objContext, true, "zone", "get", realmArg, zoneGroupArg, zoneArg)
 	if err == nil {
-		logger.Debugf("ceph zone %q already exists, new zone and pools will not be created", zone.Name)
+		logger.Debugf("ceph zone %q already exists, new zone and pools will not be created but checking for update", zone.Name)
+		if object.UpdateZoneEndpointList(zoneGroupJson.Zones, zone.Spec.CustomEndpointList, objContext.Zone) {
+			zoneEndpoints := strings.Join(zone.Spec.CustomEndpointList, ",")
+			logger.Debugf("Updating endpoints for zone %q are: %q", objContext.Zone, zoneEndpoints)
+			endpointArg := fmt.Sprintf("--endpoints=%s", zoneEndpoints)
+			err = object.JoinMultisite(objContext, endpointArg, zoneEndpoints, zone.Namespace)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
 		return reconcile.Result{}, nil
 	}
 
```

Comment:
This function is not idempotent. There is a race condition. If Rook gets to this code block, it will create the zone. If the operator stops immediately after, no pools will have been created. Then, when the operator restarts this reconcile, it will end here, and the pools will never be created. @thotz, please file this as a separate issue to fix.

Reply:
created https://github.com/rook/rook/issues/10640 to track it seperately

Reply:
Also @alimaredia please share your thoughts above issue either here or on #16040